lib/ansible/modules/web_infrastructure/ansible_tower/tower_workflow_template.py

```diff
@@ -0,0 +1,183 @@
+    name:
+      description:
+        - The name to use for the workflow.
+      required: True
+    organization:
+      description:
+        - The organization the workflow is linked to.
+      required: False
+    schema:
+      description:
+        - >
+          The schema is a JSON- or YAML-formatted string defining the
+          hierarchy structure that connects the nodes. Refer to Tower
+          documentation for more information.
+      required: False
+    survey_enabled:
+      description:
+        - Setting that variable will prompt the user for job type on the
+          workflow launch.
+      required: False
+      type: bool
+    survey:
+      description:
+        - The definition of the survey associated to the workflow.
+      required: False
+    state:
+      description:
+        - Desired state of the resource.
+      required: False
+      default: "present"
+      choices: ["present", "absent"]
+extends_documentation_fragment: tower
+'''
+
+
+EXAMPLES = '''
+- tower_workflow_template:
+    name: Workflow Template
+    description: My very first Worflow Template
+    organization: My optional Organization
+    schema: "{{ lookup(file, my_workflow.json }}"
+
+- tower_worflow_template:
+    name: Workflow Template
+    state: absent
+'''
+
+
+RETURN = ''' # '''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ansible_tower import (
+    TowerModule,
+    tower_auth_config,
+    tower_check_mode
+)
+
+try:
+    import tower_cli
+    import tower_cli.exceptions as exc
+    from tower_cli.conf import settings
+except ImportError:
+    pass
+
+
+def main():
+    argument_spec = dict(
+        name=dict(required=True),
+        description=dict(required=False),
+        extra_vars=dict(required=False),
+        organization=dict(required=False),
+        allow_simultaneous=dict(type='bool', required=False),
+        schema=dict(required=False),
+        survey=dict(required=False),
+        survey_enabled=dict(type='bool', required=False),
+        state=dict(choices=['present', 'absent'], default='present'),
+    )
+
+    module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    name = module.params.get('name')
+    state = module.params.get('state')
+
+    json_output = {'workflow_template': name, 'state': state}
+
+    tower_auth = tower_auth_config(module)
+    with settings.runtime_values(**tower_auth):
+        tower_check_mode(module)
+        wfjt_res = tower_cli.get_resource('workflow')
+        try:
+            params = {}
+            params['name'] = name
+
+            if module.params.get('description'):
+                params['description'] = module.params.get('description')
+
+            if module.params.get('organization'):
+                organization_res = tower_cli.get_resource('organization')
+                try:
+                    organization = organization_res.get(
+                        name=module.params.get('organization'))
```

Comment:
is it okay that the organization can be None here?

Reply:
Okay, so alancoding says that this cold throw an error that it can't figure out which organization you meant if there is more than one organization.

Perhaps what we need here is:
``` python
if organization is not None:
    try:
        organization = organization_res.get([....]
```

Reply:
Isn't that covered by `if module.params.get('organization'):`?

Reply:
Oops, sorry, you are correct.